lib/miq_preloader.rb

```diff
@@ -1,21 +1,53 @@
 module MiqPreloader
+  # If you want to preload an association on multiple records
+  # or want to only load a subset of an association
+  #
+  # example:
+  #   vms = Vm.where(:ems_id => emses.id)
+  #   preload_reverse(emses, :vms, vms)
+  #   emses.map { |ems| ems.vms } # cached - no queries
+  #
+  # @param record [relation|ActiveRecord::Base|Array[ActiveRecordBase]]
+  # @param association [Symbol|Hash|Array]
+  # @param preload_scope [Nil|relation] an array is not currently supported
   def self.preload(records, associations, preload_scope = nil)
     preloader = ActiveRecord::Associations::Preloader.new
     preloader.preload(records, associations, preload_scope)
   end
 
+  # If you don't use an association to load records, this method can be used
+  # to assign the parent record to the association
+  #
+  # example:
+  #   vms = Vm.where(:ems_id => ems.id)
+  #   preload_reverse(ems, :vms, vms)
```

Comment:
Is this symbol correct?  I think this is supposed to be:

```suggestion
  #   preload_reverse(ems, :ems, vms)
```

Reply:
Otherwise I don't understand this symbol